network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/rekey/NetPRekeyScheduler.kt

```diff
@@ -16,47 +16,35 @@
 
 package com.duckduckgo.networkprotection.impl.rekey
 
-import androidx.work.Constraints
-import androidx.work.ExistingPeriodicWorkPolicy
-import androidx.work.NetworkType
-import androidx.work.PeriodicWorkRequestBuilder
-import androidx.work.WorkManager
 import com.duckduckgo.app.global.DispatcherProvider
-import com.duckduckgo.appbuildconfig.api.AppBuildConfig
+import com.duckduckgo.app.utils.ConflatedJob
 import com.duckduckgo.di.scopes.VpnScope
 import com.duckduckgo.mobile.android.vpn.VpnFeaturesRegistry
-import com.duckduckgo.mobile.android.vpn.boundToVpnProcess
 import com.duckduckgo.mobile.android.vpn.service.VpnServiceCallbacks
 import com.duckduckgo.mobile.android.vpn.state.VpnStateMonitor.VpnStopReason
 import com.duckduckgo.networkprotection.impl.NetPVpnFeature
 import com.squareup.anvil.annotations.ContributesMultibinding
-import java.util.concurrent.TimeUnit.HOURS
+import java.util.concurrent.TimeUnit
 import javax.inject.Inject
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.launch
+import kotlinx.coroutines.*
 import logcat.logcat
 
 @ContributesMultibinding(VpnScope::class)
 class NetPRekeyScheduler @Inject constructor(
-    private val workManager: WorkManager,
     private val vpnFeaturesRegistry: VpnFeaturesRegistry,
     private val dispatcherProvider: DispatcherProvider,
-    private val appBuildConfig: AppBuildConfig,
+    private val netPRekeyer: NetPRekeyer,
 ) : VpnServiceCallbacks {
+
+    private val job = ConflatedJob()
+
     override fun onVpnStarted(coroutineScope: CoroutineScope) {
-        coroutineScope.launch(dispatcherProvider.io()) {
-            if (vpnFeaturesRegistry.isFeatureRegistered(NetPVpnFeature.NETP_VPN)) {
-                logcat { "NetPRekeyScheduler onVpnStarted called with NetP enabled" }
-                val constraints = Constraints.Builder()
-                    .setRequiredNetworkType(NetworkType.CONNECTED)
-                    .build()
-                val workerRequest = PeriodicWorkRequestBuilder<NetPRekeyWorker>(1, HOURS)
-                    .boundToVpnProcess(appBuildConfig.applicationId) // this worker is executed in the :vpn process
-                    .setConstraints(constraints)
-                    .build()
-                // Once work is already scheduled, we don't update the interval
-                // since onVpnStarted will be called many times when restarting/reconnecting the VPNService.
-                workManager.enqueueUniquePeriodicWork(DAILY_NETP_REKEY_TAG, ExistingPeriodicWorkPolicy.KEEP, workerRequest)
+        job += coroutineScope.launch(dispatcherProvider.io()) {
+            while (isActive && vpnFeaturesRegistry.isFeatureRegistered(NetPVpnFeature.NETP_VPN)) {
```

Comment:
Ensure we never try to rekey when device is idle.

Reply:
Question: What will happen to the previously scheduled job DAILY_NETP_REKEY_TAG after this change gets to the client? Will they automatically get cancelled?